todo_app/data/session_items.py

```diff
@@ -67,3 +67,17 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+def delete_item(itemId):
+    """
+    Deletes an existing item in the session. If no existing item matches the ID of the specified item, nothing is deleted.
+
+    Args:
+        itemId: The item to delete.
+    """
+    existing_items = get_items()
+    updated_items = list(filter(lambda item: item['id'] != itemId, existing_items))
+
+    session['items'] = updated_items
+
+    return  list(filter(lambda item: item['id'] == itemId, existing_items))[0]
```

Comment:
Nice, I'd suggest the below is slightly easier to read (avoiding the list(filter syntax being repeated)  but no strong preference:

```python
    existing_items = get_items()
    item_to_delete = next(filter(lambda item: item['id'] == itemId, existing_items))

    existing_items.remove(item_to_delete)

    session['items'] = existing_items

    return  item_to_delete
```

Worth noting that we should adjust the argument casing from itemId to item_id though

Reply:
yeah much easier to read, have made the change.